packages/dd-trace/src/encode/0.4.js

```diff
@@ -271,12 +271,45 @@ class AgentEncoder {
     }
   }
 
+  _encodeMetaStruct (bytes, value) {
+    const keys = Array.isArray(value) ? [] : Object.keys(value)
+    const validKeys = keys.filter(key =>
+      typeof value[key] === 'string' ||
+      typeof value[key] === 'number' ||
+      (value[key] !== null && typeof value[key] === 'object'))
+
+    this._encodeMapPrefix(bytes, validKeys.length)
+
+    for (const key of validKeys) {
+      this._encodeString(bytes, key)
+      this._encodeObjectAsByteArray(bytes, value[key])
+    }
```

Comment:
I'm vacillating on recommending using `Object.entries` here. On one hand, it creates more temporary small arrays. On the other hand, it eliminates the repetitive evaluation of `value[key]` and makes the filter predicate context-free, which I think are nice properties for reading comprehension of the code.

 At the end of the day, it's a matter of preference, but thought I'll raise it for consideration.

```suggestion
    const entries = Array.isArray(value) ? [] : Object.entries(value)
    const validEntries = entries.filter(([, v]) =>
      typeof v === 'string' ||
      typeof v === 'number' ||
      (v !== null && typeof v === 'object'))

    this._encodeMapPrefix(bytes, validEntries.length)

    for (const [k, v] of validEntries) {
      this._encodeString(bytes, k)
      this._encodeObjectAsByteArray(bytes, v)
    }
```

Reply:
Ok, I've done the changes. And changes also the other method that I made in the previous PR to use the same approach. thanks

Reply:
I came back to the previous approach. Even if `Object.entries` is easier to understand, it creates multiple arrays increasing memory and CPU usage.